### ISSUE TITLE
Compile with noUnusedLocals

### ISSUE THREAD

**Issue by jasongrout** [mathjax/MathJax-src#347]
Congratulations on the 3.0 release!

We're using MathJax from Typescript. Since the `mathjax` package does not ship with `.d.ts` typing files, we use the `mathjax-full` repo directly in https://github.com/jasongrout/jupyter-renderers/tree/mathjax3-final/packages/mathjax3-extension (PR at https://github.com/jupyterlab/jupyter-renderers/pull/213)

I'm running into issues when compiling with "noUnusedLocals" set to true in my tsconfig - it complains about lots of unused imports and declarations in the MathJax codebase. Perhaps you can turn it on in this repo and see if cleaning up the things complains about helps in the codebase:

```diff
diff --git a/tsconfig.json b/tsconfig.json
index e58d9955..3cc5c509 100644
--- a/tsconfig.json
+++ b/tsconfig.json
@@ -6,6 +6,7 @@
     "module": "commonjs",
     "declaration": true,
     "noImplicitAny": true,
+    "noUnusedLocals": true,
     "removeComments": true,
     "lib": ["es6", "dom"],
     "noLib": false,
```

*Edit: update commit reference*

**Comment by jasongrout** [comment]
Related to this, it would be great if a .d.ts type file could be bundled with the `mathjax` package so we could use it directly from typescript, instead of needing to use `mathjax-full`.

**Comment by dpvc** [comment]
@jasongrout, about the `.d.ts` files, see mathjax/MathJax#2197 for discussion about this and why that is not currently being done.  If you have suggestions about that, let me know.

My expectations was that typescript users would use the `mathjax-full` package.  The `mathjax` package is really for the various CDNs, as it is what CDNs like jsDelivr use to determine the files to make available.  We don't need to have `.d.ts` files available on the CDNs, for example.

-----

I've looked at the `noUnusedLocals`, but haven't had the chance to do anything about it.  Some of the imports where there because earlier versions of Typescript required them, even though the items were not used explicitly in the files; I would get errors about types not being exported or being private or some such thing if the imports were not there.  I never figured out exactly what was behind that.

Some others are due to deconstructing arrays.  For example

    const [a, b, c] = someArray;

when only `b` and `c` are used.  Do you recommend changing that to

    const [b, c] = someArray.slice(1);

or is there something better?

**Comment by jasongrout** [comment]
> `const [a, b, c] = someArray;`

For this case, you could do `const [, b, c] = someArray`.

**Comment by dpvc** [comment]
Great!  I didn't know that.

**Comment by jasongrout** [comment]
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Ignoring_some_returned_values

**Comment by dpvc** [comment]
included in v3.1 released today.
